src/macros.jl

```diff
@@ -156,29 +156,47 @@ function parse_one_operator_constraint(_error::Function, vectorized::Bool,
     return parse_code, _build_call(_error, vectorized, variable, set)
 end
 
+function rewrite_call_expression(_error::Function, head::Val{F}, args...) where F
+    _error("function $F not implemented.")
+end
+
 _functionize(v::VariableRef) = convert(AffExpr, v)
 _functionize(v::AbstractArray{VariableRef}) = _functionize.(v)
 _functionize(x) = x
 function parse_one_operator_constraint(_error::Function, vectorized::Bool, sense::Val, lhs, rhs)
+    parse_code_rhs, build_code_rhs, new_rhs = :(), :(), rhs
+    if isexpr(rhs, :call)
+        parse_code_rhs, build_code_rhs, new_rhs = rewrite_call_expression(_error, Val(rhs.args[1]), rhs.args[2:end]...)
+    end
+
     # Simple comparison - move everything to the LHS.
-    #
-    # Note: We add the +0 to this term to account for the pathological case that
-    # the `lhs` is a `VariableRef` and the `rhs` is a summation with no terms.
-    # Without the `+0` term, `aff` would evaluate to a `VariableRef` when we
-    # really want it to be a `GenericAffExpr`.
-    if vectorized
-        func = :($lhs .- $rhs)
+
+    set = sense_to_set(_error, sense)
+    # TODO: bug in MutableArithmetics? The returned code does not find $new_rhs (ERROR: UndefVarError: #642###976 not defined). 
+    if rhs == new_rhs
+        if vectorized
+            func = :($lhs .- $new_rhs)
+        else
+            func = :($lhs - $new_rhs)
+        end
+        variable, parse_code = _MA.rewrite(func)
     else
-        func = :($lhs - $rhs)
+        variable, parse_code = _MA.rewrite(lhs)
+        parse_code = :($parse_code; $variable = _MA.mutable_operate!(-, convert(AffExpr, $variable), $new_rhs))
     end
-    set = sense_to_set(_error, sense)
-    variable, parse_code = _MA.rewrite(func)
-    return parse_code, _build_call(_error, vectorized, :(_functionize($variable)), set)
+
+    build_code = _build_call(_error, vectorized, :(_functionize($variable)), set)
+    return :($parse_code_rhs; $parse_code), :($build_code_rhs; $build_code)
+end
+
+function parse_constraint(_error::Function, ::Val{:call}, args...)
+    return parse_call_constraint(_error, Val(args[1]), args[2:end]...)
```

Comment:
I would rather have this call `parse_constraint(_error, args...)` to avoid breaking existing code

Reply:
How can you then dispatch on the function being called (i.e. `args[1]`)?

Reply:
You can just keep the existing `parse_constraint(_error::Function, sense::Symbol, lhs, rhs)`, the call will be dispatched to it, isn't it ?

Reply:
Sorry, I skipped your comment… I don't see how Julia could dispatch on a symbol without Val. Or would you rather have this function called `parse_constraint`?

Reply:
It is dispatched in `parse_one_operator_constraint` as it is currently.

Reply:
Is the new version what you meant ?